### PR TITLE
RavenDB-14269: fix JsConverter strip 'Value' or 'Key' from dictionary…

### PR DESCRIPTION
**Pull request by garayx**
… entity

### CODE REVIEW

File: src/Raven.Client/Util/JavascriptConversionExtensions.cs [hunk 1]
```diff
@@ -2411,14 +2412,23 @@ private static bool CanConvert(Expression expression, DocumentConventions conven
...
-        public static Expression GetInnermostExpression(MemberExpression expression, out string path)
+        private static ParameterExpression GetParameterAndCheckInternalMemberName(MemberExpression expression, out bool hasInternal)
+        {
+            return GetInnermostExpression(expression, out _, out hasInternal) as ParameterExpression;
+        }
+
+        public static Expression GetInnermostExpression(MemberExpression expression, out string path, out bool hasInternal)
         {
             path = string.Empty;
+            hasInternal = false;
```
Comment (grisha-kotler) [inline src/Raven.Client/Util/JavascriptConversionExtensions.cs]:
```suggestion
            hasInternalKeyOrValue = false;
```

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Util/JavascriptConversionExtensions.cs (hunk 1 above)
Comment (montyclt) [reply src/Raven.Client/Util/JavascriptConversionExtensions.cs]: Caution, hasInternal variable is used too in line 265

Comment (garayx) [reply src/Raven.Client/Util/JavascriptConversionExtensions.cs]: done

**Review by grisha-kotler — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]
@aviv86 Could you have a look?

**Review by aviv86 — APPROVED** [review]

**Pull request merged**